### PR TITLE
Use datastore ancestor for relationships between entities

### PR DESCRIPTION
**Pull request by scott97**
Uses ancestors for creating the relationships between entities. This makes the code simpler, and we do not need to enforce consistency, because google datastore will do it for us.

### CODE REVIEW

File: api/handlers/annotations.go [hunk 1]
```diff
@@ -35,54 +35,113 @@ LICENSE
...
+}
+
+// FromAnnotation creates an AnnotationResult from a model.Annotation and key, formatting it according to the requested format.
+func FromAnnotation(annotation *model.Annotation, key *datastore.Key, format *api.Format) AnnotationResult {
+	var result AnnotationResult
+	if format.Requires("id") {
+		result.ID = &key.ID
+	}
+	if format.Requires("videostream_id") {
+		// result.VideoStreamID = &annotation.VideoStreamID
```
Comment (saxon-milton) [inline api/handlers/annotations.go:69]: Still need this ?

**Comment by saxon-milton** [comment]
I'm confused by the diffs, they seem very similar to another pull request you have ? If you're branching from an already existing branch (i.e. your last PR) but your pull request requests to merge into master then in the second PR (this PR) you'll also see the diffs from the first PR. This makes it really hard to see what's actually relevant to each PR. You need to ensure your target branch is set correctly e.g. second PR target branch is actually the branch you branched off of, not master. Then once the first PR is merged into master, you can adjust the target branch of the second PR.

**Review by trekhopton — APPROVED** [review]
Looks good to me, @scruzin might want to have a look at these datastore changes incorporating ancestor functionality.

**Comment by scruzin** [comment]
Just how crucial is parent key functionality? 

There is currently no FileStore implementation for parent keys, which means that clients that use this new functionality will only work with CloudStore (a.k.a. Google datastore). Perhaps this is okay, but it represents a divergence from FileStore/CloudStore compatibility to date.

That said, it probably won't be too hard to extend FileStore to represent parent keys. FileStore encodes the key in the path name, so it could optionally encode the parent key as well, separated by a character that cannot be part of a key (such as a space).

**Pull request closed without merging**